web/views.py

```
@@ -3274,7 +3274,18 @@ def current_weekly_challenge(request):
 def challenge_detail(request, challenge_id):
     try:
         challenge = get_object_or_404(Challenge, id=challenge_id)
-        submissions = ChallengeSubmission.objects.filter(challenge=challenge)
+
+        if challenge.challenge_type == "weekly":
+            # Weekly Challenge: Show only public submissions or the user's submission
+            submissions = ChallengeSubmission.objects.filter(challenge=challenge).filter(
+                Q(is_public=True) | Q(user=request.user)
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)